### ISSUE TITLE
SQLAlchemy result.rowcount is always -1

### ISSUE THREAD

**Issue by rectalogic** [aws/amazon-redshift-python-driver#73]
## Driver version
2.0.902

## Redshift version
PostgreSQL 8.0.2 on i686-pc-linux-gnu, compiled by GCC gcc (GCC) 3.4.2 20041017 (Red Hat 3.4.2-6.fc3), Redshift 1.0.33904

## Client Operating System
Ubuntu 20.04

## Python version
3.8.10

## Problem description
1. Expected behaviour:
   SQLAlchemy query results should return rowcount
2. Actual behaviour:
   -1 is returned

SQLAlchemy with `psycopg2` driver returns the correct result `rowcount`. With `redshift_connector` the rowcount is always -1
I guess this isn't always supported by drivers, is there a way to support it with `redshift_connector`?
https://docs.sqlalchemy.org/en/14/core/connections.html?highlight=rowcount#sqlalchemy.engine.BaseCursorResult.rowcount

## Reproduction code
```python
import datetime

import sqlalchemy as sa

dsn = "redshift+redshift_connector:///?..."  # returns -1
# dsn = "redshift+psycopg2:///?..."  # returns 1

engine = sa.create_engine(dsn)
statement = sa.select([1])
results = engine.execute(statement)
print(f"rowcount={results.rowcount}")
```

**Comment by Brooke-white** [comment]
hey @rectalogic,

Thanks for opening this issue. My understanding is that DB-API support for `rowcount` can be limited by the server's functionality to provide this information on the wire (i.e. if rowcount is sent to the driver or not). `redshift_connector` currently supports `rowcount` for some types statements such as `UPDATE`, `INSERT`, etc. See [here](https://github.com/aws/amazon-redshift-python-driver/blob/0d39e942be91969b24a6c8ce2f1a29df166937e7/redshift_connector/core.py#L1424) for more details.

I believe Redshift server does not provide information on `rowcount` when `SELECT` statements are executed, but I will verify this with the Redshift server team, and if it is not the case we will work to add support in `redshift_connector` to grab this information from the wire.

If `rowcount` is not supported in Redshift server, I think one way it could *maybe* be provided would be through adding logic to set in the case the type of statement is `SELECT` based off the row queue length or some sort of row counter we could add since we probably don't want to require all the rows have been processed from the wire, maybe [here](https://github.com/aws/amazon-redshift-python-driver/blob/0d39e942be91969b24a6c8ce2f1a29df166937e7/redshift_connector/core.py#L1315).

**Comment by rectalogic** [comment]
psycopg2 with Redshift is able to return rowcount for SELECT, so there may be Redshift support. Not sure exactly where psycopg2 is getting it, maybe here https://github.com/psycopg/psycopg2/blob/master/psycopg/pqpath.c#L1191

**Comment by Brooke-white** [comment]
we will work to add support for this in the new year

**Comment by Brooke-white** [comment]
We've addressed this issue in v2.0.904. `SELECT` row count can be found in Cursor.[redshift_rowcount](https://github.com/aws/amazon-redshift-python-driver/blob/master/redshift_connector/cursor.py#L129), a new method native to redshift_connector. Please feel free to open a new issue if you have further comments or questions.
